### PR TITLE
Add unit tests for gradual fetching during VMR sync

### PR DESCRIPTION
**Pull request by premun**
Adds missing tests and fixes some edge cases missed in https://github.com/dotnet/arcade-services/issues/3080

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [x] Internal Infrastructure Improvements

### Release Note Description
Not release notes worthy

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/RepositoryCloneManager.cs [hunk 1]
```diff
@@ -9,6 +9,7 @@
 using Microsoft.DotNet.Darc.Models.VirtualMonoRepo;
 using Microsoft.DotNet.DarcLib.Helpers;
 using Microsoft.Extensions.Logging;
+using Microsoft.TeamFoundation.Build.WebApi;
```
Comment (oleksandr-didyk) [inline src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/RepositoryCloneManager.cs:12]: Is this import needed here?

File: src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/RepositoryCloneManager.cs [hunk 2]
```diff
@@ -163,15 +169,16 @@ public class RepositoryCloneManager : IRepositoryCloneManager
                 catch
                 {
                     // Ref not found yet, let's try another remote
-                    continue;
-                }
-
-                if (refsToVerify.Count == 0)
-                {
-                    _logger.LogDebug("All requested refs ({refs}) found in {repo}", string.Join(", ", requestedRefs), path);
+                    missingCommit = true;
```
Comment (tkapin) [inline src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/RepositoryCloneManager.cs:172]: Should we log what commit is missing? As we log a message when all commits are found below, it would be useful to also log the complementary information about what's missing.

File: test/Microsoft.DotNet.DarcLib.Tests/VirtualMonoRepo/RepositoryCloneManagerTests.cs [hunk 3]
```diff
@@ -184,4 +186,140 @@ void ResetCalls()
...
+        }
+    }
+
+    private class RemoteState
+    {
+        public string RemoteUri { get; set; }
+
+        public IReadOnlyCollection<string> CommitContained { get; set; }
+
+        //public TaskCompletionSource<string> CloneTask { get; } = new();
```
Comment (oleksandr-didyk) [inline test/Microsoft.DotNet.DarcLib.Tests/VirtualMonoRepo/RepositoryCloneManagerTests.cs:313]: Nit: these commented lines should probably be removed

**Review by tkapin — APPROVED** [review]

**Review by oleksandr-didyk — APPROVED** [review]

**Comment by premun** [comment]
@oleksandr-didyk I made a small change in the last commit. PTAL

**Review by oleksandr-didyk — DISMISSED** [review]

File: src/Microsoft.DotNet.Darc/DarcLib/LocalGitClient.cs [hunk 4]
```diff
@@ -10,6 +10,7 @@
 using System.Threading.Tasks;
 using Microsoft.DotNet.DarcLib.Helpers;
 using Microsoft.Extensions.Logging;
+using Microsoft.TeamFoundation.SourceControl.WebApi;
```
Comment (oleksandr-didyk) [inline src/Microsoft.DotNet.Darc/DarcLib/LocalGitClient.cs:13]: Nit: the WebApi import is back

Comment (premun) [reply src/Microsoft.DotNet.Darc/DarcLib/LocalGitClient.cs:13]: Jesus, how??

Thanks

**Review by oleksandr-didyk — APPROVED** [review]

File: src/Microsoft.DotNet.Darc/DarcLib/LocalGitClient.cs (hunk 4 above)
Comment (oleksandr-didyk) [reply src/Microsoft.DotNet.Darc/DarcLib/LocalGitClient.cs]: I noticed Visual Studio really like to put it in for some reason, maybe some suggestion that got auto-accepted and it brough the import? Don't know
